livekit-android-sdk/src/main/java/io/livekit/android/e2ee/E2EEManager.kt

```diff
@@ -0,0 +1,182 @@
+package io.livekit.android.e2ee
+
+import io.livekit.android.events.RoomEvent
+import io.livekit.android.room.Room
+import io.livekit.android.room.participant.*
+import io.livekit.android.room.track.LocalAudioTrack
+import io.livekit.android.room.track.LocalVideoTrack
+import io.livekit.android.room.track.RemoteAudioTrack
+import io.livekit.android.room.track.RemoteVideoTrack
+import io.livekit.android.room.track.Track
+import io.livekit.android.room.track.TrackPublication
+import io.livekit.android.util.LKLog
+import org.webrtc.FrameCryptor
+import org.webrtc.FrameCryptor.FrameCryptionState
+import org.webrtc.FrameCryptorAlgorithm
+import org.webrtc.FrameCryptorFactory
+import org.webrtc.RtpReceiver
+import org.webrtc.RtpSender
+
+class E2EEManager
+constructor(keyProvider: KeyProvider)  {
+    private var room: Room? = null
+    private var keyProvider: KeyProvider
+    private var frameCryptors = mutableMapOf<String, FrameCryptor>()
+    private var algorithm: FrameCryptorAlgorithm = FrameCryptorAlgorithm.AES_GCM
+    private lateinit var emitEvent: (roomEvent: RoomEvent) -> Unit?
+    var enabled: Boolean = false
+    init {
+        this.keyProvider = keyProvider
+    }
+
+    suspend fun setup(room: Room, emitEvent: (roomEvent: RoomEvent) -> Unit) {
```

Comment:
Looks like these can be put in the constructor?

Reply:
It was discussed with Lukas here that we can enter a certain room without enabling e2ee, and use setup to switch to encrypted mode halfway, so I suggest to keep